14_more_sinatra/badreads/app/controllers/genres_controller.rb

```diff
@@ -0,0 +1,45 @@
+class GenresController < ApplicationController
+
+  get "/genres" do
+    @genres = Genre.all
+    erb :'genres/index'
+  end
+
+  post "/genres" do
+    @genre = Genre.create(name: params[:name])
+    @genre.book_ids = params[:book_ids]
+    redirect "/genres/#{@genre.id}"
```

Comment:
See `genres/new.erb` to really see why this works. Rack does a lot of the work for us :)